### ISSUE TITLE
Parity: CanvasScreen UI - dedicated composable for canvas navigation and rendering

### ISSUE THREAD

**Issue by yuga-hashimoto** [yuga-hashimoto/OpenClawAssistant#196]
## Summary

CanvasController (functional layer) is fully implemented. What's missing is the **UI layer** — the `CanvasScreen.kt` Compose composable that hosts the WebView and wires it to `CanvasController`.

## Current State

| Component | Status |
|---|---|
| `node/CanvasController.kt` (navigate, eval, snapshot, A2UI) | ✅ Complete |
| `node/InvokeDispatcher.kt` (canvas.* routing) | ✅ Complete |
| `ui/CanvasScreen.kt` composable | ❌ Missing |
| Tab navigation entry point to CanvasScreen | ❌ Missing (#203) |
| Gateway auth headers in WebView | ❌ Missing (#202) |

### `ui/CanvasScreen.kt`

```kotlin
@Composable
fun CanvasScreen(
    canvasController: CanvasController,
    modifier: Modifier = Modifier
) {
    AndroidView(
        modifier = modifier.fillMaxSize(),
        factory = { context ->
            WebView(context).also { webView ->
                webView.settings.apply {
                    javaScriptEnabled = true
                    domStorageEnabled = true
                    allowFileAccess = true
                    mixedContentMode = WebSettings.MIXED_CONTENT_COMPATIBILITY_MODE
                }
                webView.webViewClient = CanvasWebViewClient(canvasController)
                canvasController.attach(webView)
            }
        },
        onRelease = { webView ->
            webView.stopLoading()
            canvasController.detach()
            webView.destroy()
        }
    )
}
```

### `CanvasWebViewClient`

Needed for:
- Calling `canvasController.onPageFinished()` (debug status overlay)
- Injecting Gateway auth headers for resource requests (see #202)

```kotlin
class CanvasWebViewClient(
    private val canvasController: CanvasController
) : WebViewClient() {

    override fun onPageFinished(view: WebView, url: String) {
        super.onPageFinished(view, url)
        canvasController.onPageFinished()
    }

    override fun shouldInterceptRequest(
        view: WebView,
        request: WebResourceRequest
    ): WebResourceResponse? {
        // Auth header injection — see #202
        return null
    }
}
```

### Add `detach()` to `CanvasController`

Currently `CanvasController` only has `attach(webView)`. A matching `detach()` is needed for cleanup on `onRelease`:

```kotlin
fun detach() {
    webView = null
}
```

### Wire into `MainViewModel`

`CanvasController` is currently instantiated somewhere in the node layer. `CanvasScreen` needs a reference to it — either via `MainViewModel` or by DI.

## Acceptance Criteria

- [ ] `CanvasScreen` composable exists in `ui/CanvasScreen.kt`
- [ ] WebView is configured: JS enabled, DOM storage enabled, file access allowed
- [ ] `CanvasController.attach(webView)` called on WebView creation
- [ ] `CanvasController.detach()` / `webView.destroy()` called on disposal
- [ ] `canvasController.onPageFinished()` called from `WebViewClient.onPageFinished()`
- [ ] `CanvasController` is accessible from `CanvasScreen` (via ViewModel or direct reference)
- [ ] CanvasScreen is reachable from main UI (see #203 for navigation)

## Related Issues

- #202 — Gateway auth headers in WebView (implement inside `CanvasWebViewClient`)
- #203 — Compose Navigation (needed to reach CanvasScreen from main UI)
